### ISSUE TITLE
30-Compositional Text-to-Image Synthesis with Attention Map Control of Diffusion Models

### ISSUE THREAD

**Issue by mjmaher987** [sharif-ml-lab/visual_compositional_generation_survey#30]
## Task: Read the Paper and Do the following 4 steps
1. Highlight the paper systematically (**green** for important notes - **blue** for very important notes that worth mentioning in the final survey - **red** for the sentences that we don’t understand or need further attention in the future) [In Zotero].
2. Write the Wikipedia summary of the paper in the GitHub issue (_comment in this issue_).
3. Extract the information in the [excel file](https://docs.google.com/spreadsheets/d/15_sh6TqtUvZIJo1-RZxQtsG0Nvg0vPG_IFKCWiGMjdY/edit?usp=sharing).
4. Add the related sentences to the [overleaf file](https://www.overleaf.com/8167246262ccxtrzztjvpz). Note that when you add the sentence, don’t forget to write the reference in the format of ”[ #reference number that is mentioned in the Google Doc - Name of the Author (who added this to the overleaf) ]” at the end of the sentence.

**Comment by AmirMohammadFakhimi** [comment]
# Compositional Text-to-Image Synthesis with Attention Map Control of Diffusion Models

## Problems
Attribute leakage, entity leakage, and missing entities. The paper divides all defects of diffusion models in these categories.

## Method:
<img width="713" alt="Method" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/3f36128e-5131-4f83-9c3b-0bb91ebfddcd">

1. **BoxNet:** They propose BoxNet, an object box prediction module capable of estimating object locations at any time step during the forward diffusion process. At time step $t$, there's a noisy image $z_t$. It is one of the inputs of the BoxNet. More accurately, it is the input of the U-Net inside of the BoxNet. They extract the output feature maps from each down- and up-sampling layer of the U-Net. All the extracted feature maps are interpolated into the same size and concatenated together. A linear transformation is then applied to acquire a feature tensor $f$ that represents the current denoised latent $z_t$. After that, we use a standard encoder-decoder transformer to generate entity boxes. Each box is consist of $(c_x, c_y, w, h)$ where $c_x$ and $c_y$ means the center coordinate of the box and $w$, $h$ means the width and height of the box. Note that the encoder expects a sequence as input, hence we reshape the spatial dimensions of f into one dimension. The decoder decodes boxes with input entity queries. To acquire entity queries, the text prompt input by a user is first parsed into $N$ entities with attributes manually or by an existing text parser. Then, the entity phrases are encoded into embeddings by the text encoder. Entity embeddings are pad with a trainable placeholder tensor into max length $M$ and only the first $N$ of the output sequences are used to calculate entity boxes by a weight shared linear projection layer.  
As to the training phase, we train the BoxNet in the forward process of SD on the COCO dataset.  
**Loss:** They first produce an optimal bipartite matching between predicted and ground truth boxes, and then they optimize entity box losses. Let's denote by $b$ the ground truth set of $N$ objects, and $b′$ the set of top $N$ predictions. To find a bipartite matching between these two sets we search for a permutation of $N$ elements $\sigma \in P_N$ with the lowest cost:
    <img width="249" alt="Loss 1" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/479f2f67-5a17-4a70-a579-3acaba16e5c9">

    They prioritize classification accuracy over location accuracy by modifying the matching cost to include an extremely high penalty for bounding boxes with class mismatches ($\lambda$ is a high value number and $\lambda_{iou}$, $\lambda_{L1}$ are hyperparameters.):
    <img width="382" alt="Loss 2" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/f153ece0-887d-438c-95e4-a4e01c0c62aa">
    <img width="388" alt="Loss 3" src="https://github.com/sharif-ml-lab/visual_compositional_generation_survey/assets/79265203/0f89372e-4a2b-47e3-a4df-11de5a1144ad">

     BoxNet generalize well to unseen entities beyond COCO dataset.

2. We develop an effective attention mask control strategy based on the proposed BoxNet, which constrains the attention areas to lie within the predicted boxes. • The trained BoxNet and attention mask control of our method can be easily incorporated into existing diffusion-based generators as a ready-to-use plugin. We demonstrate our model’s capability by integrating it into two existing models, Attend-and-Excite [11] and GLIGEN [7].
